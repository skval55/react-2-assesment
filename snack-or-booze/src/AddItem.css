.AddItem {
    width: 75%;
    max-width: 500px;
    text-align: center;
    padding: 10px;
    background-color: #101010d1;
    color: #f0f2f5;
    border: 20px solid rgba(16, 16, 16, 0.63);
    box-shadow: inset 0 0 0 2px #ffffff;
    box-sizing: border-box;
}

.AddItem-form p{
    color: rgb(155, 65, 65);
}
.AddItem-form legend{
    color: rgb(168, 168, 168);
}
.AddItem-form legend:focus{
    color: rgb(168, 168, 168);
}
.AddItem-form div label{
    color: rgb(168, 168, 168);
}
.AddItem-form div input{
    color: rgb(168, 168, 168);
}
.AddItem-form button{
    color: rgb(168, 168, 168);
}