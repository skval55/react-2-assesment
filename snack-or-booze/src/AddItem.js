import React, { useState } from "react";
import { useHistory } from "react-router-dom";

const AddItem = ({ addSnack, addDrink }) => {
  const INITIAL_STATE = {
    snackOrDrink: "",
    id: "",
    name: "",
    description: "",
    recipe: "",
    serve: "",
  };

  const [validated, setValidated] = useState(false);

  const history = useHistory();

  const [formData, setFormData] = useState(INITIAL_STATE);

  const validateForm = (formData) => {
    for (let key in formData) {
      if (formData[key] === "") {
        return false;
      }
    }
    return true;
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (validateForm(formData)) {
      if (formData.snackOrDrink === "snack") {
        addSnack(formData);
        setFormData(INITIAL_STATE);
        history.push("/snacks");
      } else if (formData.snackOrDrink === "drink") {
        addDrink(formData);
        setFormData(INITIAL_STATE);
        const addedDrinks = localStorage.getItem("drinks");
        console.log(addedDrinks);
        localStorage.setItem(
          "drinks",
          JSON.stringify([...JSON.parse(addedDrinks), formData])
        );
        console.log(localStorage.getItem("drinks"));
        history.push("/drinks");
      }
    } else {
      setValidated(true);
    }
  };

  const handleChange = (e) => {
    let { name, value } = e.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
    console.log(formData);
  };

  return (
    <form onSubmit={handleSubmit}>
      {validated ? <p>Please fill every value</p> : null}
      <p>Snack or Drink</p>
      <input
        type="radio"
        id="snack"
        name="snackOrDrink"
        value="snack"
        onChange={handleChange}
      />
      <label htmlFor="snack">Snack</label> <br />
      <input
        type="radio"
        id="drink"
        value="drink"
        name="snackOrDrink"
        onChange={handleChange}
      />
      <label htmlFor="drink">Drink</label> <br />
      <label htmlFor="id">Id</label>
      <input id="id" name="id" onChange={handleChange} /> <br />
      <label htmlFor="name">Name</label>
      <input id="name" name="name" onChange={handleChange} /> <br />
      <label htmlFor="description">Description</label>
      <input id="description" name="description" onChange={handleChange} />{" "}
      <br />
      <label htmlFor="recipe">Recipe</label>
      <input id="recipe" name="recipe" onChange={handleChange} /> <br />
      <label htmlFor="serve">How to Serve</label>
      <input id="serve" name="serve" onChange={handleChange} /> <br />
      <button>Add</button>
    </form>
  );
};

export default AddItem;
