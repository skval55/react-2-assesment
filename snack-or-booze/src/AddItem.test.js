import { render, screen } from "@testing-library/react";
import AddItem from "./AddItem";
import React from "react";

// smoke test see if it renders
test("see if it renders", () => {
  render(
    <AddItem
      addSnack={() => console.log("test")}
      addDrink={() => console.log("test")}
    />
  );
});
test("see if it renders", () => {
  const { asFragment } = render(
    <AddItem
      addSnack={() => console.log("test")}
      addDrink={() => console.log("test")}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
