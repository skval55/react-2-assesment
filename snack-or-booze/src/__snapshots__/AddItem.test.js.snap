// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`see if it renders 1`] = `
<DocumentFragment>
  <div
    class="AddItem"
  >
    <div
      class="MuiFormControl-root AddItem-form MuiFormControl-fullWidth"
    >
      <legend
        class="MuiFormLabel-root MuiFormLabel-root"
      >
        Snack or Drink
      </legend>
      <div
        aria-label="snackOrDrink"
        class="MuiFormGroup-root"
        role="radiogroup"
      >
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-8 MuiRadio-root"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-11"
                name="snackOrDrink"
                type="radio"
                value="snack"
              />
              <div
                class="PrivateRadioButtonIcon-root-12"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-13"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Snack
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-8 MuiRadio-root"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-11"
                name="snackOrDrink"
                type="radio"
                value="drink"
              />
              <div
                class="PrivateRadioButtonIcon-root-12"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-13"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Drink
          </span>
        </label>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="id"
          id="id-label"
        >
          Id
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="id"
            name="id"
            type="text"
            value=""
          />
        </div>
      </div>
       
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="name"
          id="name-label"
        >
          Name
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="name"
            name="name"
            type="text"
            value=""
          />
        </div>
      </div>
       
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="description"
          id="description-label"
        >
          Description
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="description"
            name="description"
            type="text"
            value=""
          />
        </div>
      </div>
       
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="recipe"
          id="recipe-label"
        >
          Recipe
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="recipe"
            name="recipe"
            type="text"
            value=""
          />
        </div>
      </div>
       
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="serve"
          id="serve-label"
        >
          How to serve
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="serve"
            name="serve"
            type="text"
            value=""
          />
        </div>
      </div>
       
      <br />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Add
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
